src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -63,6 +64,20 @@ public EntityInteractHandler getItemHandler() {
                     zombieVillager.setConversionPlayer(p);
                 }
             }
+            else if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16)
+                    && entity.getType() == EntityType.ZOMBIFIED_PIGLIN) {
+                Player p = e.getPlayer();
+
+                if (p.getGameMode() != GameMode.CREATIVE) {
+                    ItemUtils.consumeItem(item, false);
+                }
+
+                Location loc = entity.getLocation();
+                p.playSound(loc, Sound.ENTITY_ZOMBIE_VILLAGER_CONVERTED, 1, 1);
+
+                entity.remove();
+                loc.getWorld().spawnEntity(loc, EntityType.PIGLIN);
+            }
```

Comment:
This is pretty redundant with the above, there's probably lots of ways to refactor this.